code-generators/java-types/buildSrc/src/main/java/uk/gov/di/codegen/CustomSchemaRule.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.di.codegen;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.sun.codemodel.JClassContainer;
+import com.sun.codemodel.JType;
+import org.jsonschema2pojo.Schema;
+import org.jsonschema2pojo.rules.RuleFactory;
+import org.jsonschema2pojo.rules.SchemaRule;
+
+/**
+ * Custom schema rule that generates all classes contained within the '$defs' section of the json-schema.
+ * This may be removed once https://github.com/joelittlejohn/jsonschema2pojo/pull/1523 (or equivalent) is available.
+ */
```

Comment:
To elaborate a little more - generating from the `$defs` section means we can generate all the types from a single schema file, which avoids creating duplicate versions of the same classes.